library/std/src/io/util.rs

```diff
@@ -268,3 +268,154 @@ impl fmt::Debug for Sink {
         f.debug_struct("Sink").finish_non_exhaustive()
     }
 }
+
+/// A writer which will move data into the void, and a reader which
+/// is always at EOF. Effectively a combination of [`Sink`] and [`Empty`],
+/// similiar to `/dev/null` on Unix.
+///
+/// This struct is generally created by calling [`null()`]. Please
+/// see the documentation of [`null()`] for more details.
+#[unstable(feature = "io_null", issue = "none")]
+#[non_exhaustive]
+#[derive(Copy, Clone, Default)]
+pub struct Null;
+
+/// Creates an instance of a writer which will successfully consume all data, as
+/// well as a reader which is always at EOF.
+///
+/// All calls to [`write`] on the returned instance will return `Ok(buf.len())`
+/// and the contents of the buffer will not be inspected.
```

Comment:
This could be construed to mean that passing uninitialized memory is ok. I think we should remove the second guarantee.
```suggestion
/// All calls to [`write`] on the returned instance will return `Ok(buf.len())`.
```

Reply:
oh, I see - I probably shouldn't be the one making that decision, but it's a good callout for @joshtriplett.

Reply:
I've changed, but will change it back if recommended by someone else. 😄

Reply:
@vidhanio Please do keep the comment, to match the types being merged.